core/modules/file/file.admin.inc

```diff
@@ -585,6 +585,185 @@ function file_settings_form_validate(&$form, &$form_state) {
+      'class' => array($default_dir, 'default-' . $default_dir),
+    );
+    $rows[$directory]['data']['used_by'][] = '<strong>' . t('This is the default/root directory for %stream', array('%stream' => $stream_wrappers[$default_dir]['name'])) . '</strong>';
+  }
+
+  // Collect file upload directories as they have been configured in the various
+  // field instancess, across all entity types.
+  $file_types = array(
+    'file' => t('files'),
+    'image' => t('images'),
+  );
+
+  foreach ($instances as $entity_type => $type_bundles) {
+    foreach ($type_bundles as $bundle => $bundle_instances) {
+      foreach ($bundle_instances as $field_name => $instance) {
+        $field = field_info_field($field_name);
+
+        // We only need image and file fields.
+        if (in_array($field['module'], array('image', 'file'))) {
+          $scheme = $field['settings']['uri_scheme'];
+          $realpath = file_stream_wrapper_get_instance_by_scheme($scheme)->realpath();
+          if (!empty($instances[$entity_type][$bundle][$field_name]['settings']['file_directory'])) {
+            $realpath .= '/' . $instances[$entity_type][$bundle][$field_name]['settings']['file_directory'];
+          }
+          $directory = $realpath;
+
+          // Initialize the row if we encounter the directory for the first
+          // time.
+          if (!isset($rows[$directory])) {
+            $rows[$directory] = array(
+              'data' => array(
+                'dir' => $directory,
+                'scheme' => $stream_wrappers[$scheme]['name'],
+                'used_by' => array(),
+              ),
+              'class' => array($scheme),
+            );
+          }
+
+          // Add the current instance.
+          $admin_path = _field_ui_bundle_admin_path($entity_type, $bundle);
+          $entity_type_label = strtolower($entity_types[$entity_type]['label']);
+          $bundle_label = $bundles[$entity_type][$bundle]['label'];
+          $bundle_manage_fields_link = $admin_path . '/fields';
+          $field_label = $instances[$entity_type][$bundle][$field_name]['label'];
+          $field_name = $instances[$entity_type][$bundle][$field_name]['field_name'];
+          $field_configure_link = $bundle_manage_fields_link . '/' . $field_name;
+          $file_type = $file_types[$field['module']];
+
+          $t_vars = array(
+            '@file_type' => $file_type,
+            '@field_configure_link' => url($field_configure_link),
+            '%field_label' => $field_label,
+            '@entity_type' => $entity_type_label,
+            '@bundle_manage_fields_link' => url($bundle_manage_fields_link),
+            '%bundle_label' => $bundle_label,
+          );
+          if ($admin_path) {
+            $rows[$directory]['data']['used_by'][] = t('@file_type uploaded in the <a href="@field_configure_link">%field_label</a> field in the @entity_type <a href="@bundle_manage_fields_link">%bundle_label</a>', $t_vars);
+          }
+          else {
```

Comment:
Any idea, when this "else" strikes? (Just a question, no review. :wink: )

Reply:
Not 100% sure, but TBH all this logic here is mostly copy-paste from https://github.com/backdrop/backdrop/blob/1.x/core/modules/field_ui/field_ui.admin.inc#L53